examples/gno.land/p/demo/zteams/teams.gno

```diff
@@ -0,0 +1,262 @@
+package zteams
+
+import (
+    "std"
+    "gno.land/p/demo/avl"
+    "gno.land/p/demo/users"
+)
+
+// Team represents a team with its details and roles.
+type Team struct {
+    Address     std.Address
+    TeamName    string
+    TeamProfile string
+    Owner       std.Address
+    Admins      map[string]bool // Changed to string keys
+    Disbursers  map[string]bool // Changed to string keys
+}
+
+// TeamManager manages teams and their members.
+type TeamManager struct {
+    Teams   *avl.Tree // Maps team address to Team details
+    Members *avl.Tree // Maps team address to a slice of user pointers
+}
+
+// NewTeamManager initializes a new TeamManager.
+func NewTeamManager() *TeamManager {
+    return &TeamManager{
+        Teams:   avl.NewTree(),
+        Members: avl.NewTree(),
+    }
+}
+
+// Helper function to retrieve a team and check for errors.
+func (tm *TeamManager) getTeam(teamAddress std.Address) (*Team, error) {
+    team, found := tm.Teams.Get(teamAddress.String())
+    if !found {
+        return nil, ErrTeamNotFound
+    }
+    return team.(*Team), nil
+}
+
+// Helper function to check if a user is the Owner of a team.
+func (tm *TeamManager) isOwner(team *Team, userAddress std.Address) bool {
+    return team.Owner == userAddress
+}
+
+// Helper function to check if a user is an Admin of a team.
+func (tm *TeamManager) isAdmin(team *Team, userAddress std.Address) bool {
+    return team.Admins[userAddress.String()]
+}
```

Comment:
The `isOwner` and `isAdmin` methods are implemented as methods of `TeamManager`. But these methods only use data from the `Team` type. So, it seems better to change them to methods of `Team` struct.

Reply:
changed the receiver to be `Team` and removed the redundant `team *Team` param.